src/nvim/ex_cmds.c

```diff
@@ -5864,3 +5928,204 @@ void set_context_in_sign_cmd(expand_T *xp, char_u *arg)
     }
   }
 }
+
+/// Open a window for future displaying of the inc_sub mode.
+///
+/// Does not allow editing in the window.
+/// Returns when the window is closed.
+///
+/// @param pat    The pattern word
+/// @param sub    The replacement word
+/// @param lmatch The list containing our data
+void ex_window_inc_sub(char_u * pat,
+                       char_u * sub,
+                       klist_MatchedLine *lmatch,
+                       bool split)
+  FUNC_ATTR_NONNULL_ARG(1, 2, 3)
+{
+  garray_T winsizes;
+  int save_restart_edit = restart_edit;
+  int save_State = State;
+  int save_exmode = exmode_active;
+  int save_cmdmsg_rl = cmdmsg_rl;
+
+  // Can't do this recursively.  Can't do it when typing a password.
+  if (cmdline_star > 0) {
+    beep_flush();
+    return;
+  }
+
+  // Save current window sizes.
+  win_size_save(&winsizes);
+
+  // Save the current window to restore it later
+  win_T *oldwin = curwin;
+
+  if (split) {
+    // Don't execute autocommands while creating the window.
+    block_autocmds();
```

Comment:
Hmm there are 2 unblocks for this block below, wonder why this did not wreak havoc... thanks!